CONFIGURATION.md

```diff
@@ -113,6 +113,8 @@ offset_commit_cb                         |  C  |                 |
 enable.partition.eof                     |  C  | true, false     |         false | low        | Emit RD_KAFKA_RESP_ERR__PARTITION_EOF event whenever the consumer reaches the end of a partition. <br>*Type: boolean*
 check.crcs                               |  C  | true, false     |         false | medium     | Verify CRC32 of consumed messages, ensuring no on-the-wire or on-disk corruption to the messages occurred. This check comes at slightly increased CPU usage. <br>*Type: boolean*
 client.rack                              |  *  |                 |               | low        | A rack identifier for this client. This can be any string value which indicates where this client is physically located. It corresponds with the broker config `broker.rack`. <br>*Type: string*
+transactional.id                         |  P  |                 |               | high       | The TransactionalId to use for transactional delivery. This enables reliability semantics which span multiple producer sessions since it allows the client to guarantee that transactions using the same TransactionalId have been completed prior to starting any new transactions. If no TransactionalId is provided, then the producer is limited to idempotent delivery. Note that enable.idempotence must be enabled if a TransactionalId is configured. The default is null, which means transactions cannot be used. Note that, by default, transactions require a cluster of at least three brokers which is the recommended setting for production; for development you can change this, by adjusting broker setting transaction.state.log.replication.factor. <br>*Type: string*
```

Comment:
> Note that enable.idempotence must be enabled if a TransactionalId is configured.

this is either not how it works (i've been apparently using transactions without setting enable.idemptence) or you should return an error on incorrect configuration to alert the user there's a problem.

----

Also, suggested alternate text (i think clearer):

"The transactional.id is used to identify the same producer instance across process restarts. It allows the client to guarantee that transactions corresponding earlier instances of the client have been finalized prior to starting any new transactions, and that any zombie instances are fenced off. If no transactional.id is provided, then the producer is limited to idempotent delivery. Note that by default, transactions require a cluster of at least three brokers which is the recommended setting for production; for development you can change this by adjusting broker setting transaction.state.log.replication.factor."

Reply:
This was copied verbatim from the Java client and subject for rewrite, thanks for beating me to it!

And yes, enable.idempotence is set automatically when transactional.id is set.